src/main/java/org/broadinstitute/hellbender/tools/dataflow/transforms/bqsr/BaseRecalibratorFn.java

```diff
@@ -81,167 +59,118 @@
 
     private RecalibrationEngine recalibrationEngine;
 
+    private boolean firstInBundle = true;
     private int minimumQToUse;
-
-    protected static final String NO_DBSNP_EXCEPTION = "This calculation is critically dependent on being able to skip over known variant sites. Please provide a VCF file containing known sites of genetic variation.";
+    private long nReadsProcessed = 0;
 
     private BAQ baq; // BAQ the reads on the fly to generate the alignment uncertainty vector
-    private ReferenceDataSource referenceDataSource; // datasource for the reference. We're using a different one from the engine itself to avoid messing with its caches.
-    private static final byte NO_BAQ_UNCERTAINTY = (byte) '@';
-
+    private final static byte NO_BAQ_UNCERTAINTY = (byte) '@';
 
-    private long accumulator;
+    // -------------------------------------------------------------
+    // public functions (and constructors, those go first)
 
-    /**
-     * This method returns a new BaseRecalibratorWorker that is configured according to the command-line
-     * arguments in "toolArgs" and the input file header in "header". The command-line parser will print
-     * any diagnostic messages to "out". If the command-line requested the program not be run
-     * (e.g. it was "--help"), then this method returns null.
-     */
-    public static BaseRecalibratorWorker fromCommandLine(SAMFileHeader header, String commandLine, PrintStream out) {
-        BaseRecalibratorWorker br = new BaseRecalibratorWorker(header);
-        br.BRAC = new BaseRecalibrationArgumentCollection();
-        boolean shouldRun = new CommandLineParser(br).parseArguments(out, Utils.escapeExpressions(commandLine));
-        if (!shouldRun) return null;
-        return br;
-    }
-
-    public static BaseRecalibratorWorker fromArgs(SAMFileHeader header, BaseRecalibrationArgumentCollection toolArgs) {
-        return new BaseRecalibratorWorker(header, toolArgs);
+    public BaseRecalibratorFn(PCollectionView<SAMFileHeader> headerView, PCollectionView<SAMSequenceDictionary> referenceSequenceDictionaryView, BaseRecalibrationArgumentCollection BRAC) {
+        super("BaseRecalibratorFn");
+        this.headerView = headerView;
+        this.referenceSequenceDictionaryView = referenceSequenceDictionaryView;
+        this.BRAC = BRAC;
     }
 
-    private BaseRecalibratorWorker(SAMFileHeader samHeader) {
-        this.samHeader = samHeader;
+    private BaseRecalibratorFn(SAMFileHeader header, BaseRecalibrationArgumentCollection BRAC) {
+        super("BaseRecalibratorFn");
+        this.header = header;
+        this.BRAC = BRAC;
```

Comment:
Add disclaimer that this constructor intentionally leaves `referenceSequenceDictionaryView` uninitialized.